test/verify/check-login

```diff
@@ -561,5 +561,135 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
+
+            output = m.execute("{} --user {}".format(module, user))
+
+            if module == 'faillock':
+                n_lines = len(output.splitlines())
+                if n_lines:
+                    n_lines -= 2 # remove the header, if it's there
+
+                self.assertEqual(expected, n_lines)
+
+            else: # pam_tally2
+                self.assertIn(' {} '.format(expected), output)
+
+            # make sure the above command didn't change the file
+            if must_exist:
+                self.assertEqual(cksum, m.execute("cksum {}".format(logfile(user))))
+
+        def expect_failed_login(user, password, n_failed):
+            b.try_login(user, password)
+            b.wait_text_not("#login-error-message", "")
+            expect_fail_count(n_failed, user=user)
+
+        def expect_successful_login(user, password):
+            b.login_and_go('/system', user=user, password=password)
+            expect_fail_count(0, user=user)
+            b.logout()
+
+        # ensure we have no module in our pam config already
+        m.execute("! grep -r '{}' /etc/pam.d".format(module))
+
+        # add it to the pam config
+        if module == 'pam_tally2':
+            self.restore_file("/etc/pam.d/cockpit")
+            m.execute("sed -i '/common-auth/ {"
+                      "iauth required pam_tally2.so deny=4\n"
+                      "}' /etc/pam.d/cockpit")
+        else:
+            self.restore_file("/etc/pam.d/password-auth")
+            m.execute("sed -i '/pam_unix/ {"
+                      "s/sufficient/[success=1 default=ignore]/\n"
+                      "aauth [default=die] pam_faillock.so authfail audit deny=4 unlock_time=600\n"
+                      "aauth sufficient pam_faillock.so authsucc audit deny=4 unlock_time=600\n"
+                      "}' /etc/pam.d/password-auth")
+
+        m.execute("grep -r '{}' /etc/pam.d".format(module))
+
+        m.start_cockpit()
+
+        # pam_faillock tries to chown the file to be owned by the user itself,
+        # which is currently an selinux fail on the following distros:
+        if m.image in ['centos-8-stream', 'fedora-31', 'fedora-32', 'rhel-8-2', 'rhel-8-2-distropkg', 'rhel-8-3']:
+            # https://bugzilla.redhat.com/show_bug.cgi?id=1836182
+            self.allow_journal_messages('.*type=1400.*avc:  denied  { chown }.*comm="cockpit-session".*')
+
+        if module == 'pam_tally2':
+            # at first, we should have no log file
+            m.execute("! test -f {}".format(logfile('admin')))
+
+        # make sure we can still login
+        b.login_and_go("/system")
+        b.logout()
+
+        if module == 'pam_tally2':
+            # now we should have it
+            m.execute("test -f {}".format(logfile('admin')))
+
+        # and it should show zero fails
+        expect_fail_count(0, must_exist=False)
+
+        # try three bogus login attempts
+        for n in [1, 2, 3]:
+            expect_failed_login("admin", "bad", n)
```

Comment:
After that it could call `expect_fail_count(3)` to make sure that the tallying works? Mostly to fail earlier and make it more obvious what to debug? Just a thought, feel free to ignore.

Reply:
Added now.